docs/example_go.md

```diff
@@ -3,29 +3,273 @@ title: "Example: Go"
+
+To skip straight to the fully optimized setup, go to this subdirectory:
+
+[Recommended Tiltfile](https://github.com/windmilleng/tilt-example-go/blob/master/3-recommended/Tiltfile)
+
+## Step 0: The Simplest Deployment
 
-## Servantes
+Our server is a simple HTML server that uses Gorilla Mux for routing requests:
 
-[Servantes](https://github.com/windmilleng/servantes) is a personalized homepage
-app implemented with many microservices in multiple languages. You can run it
-with the commands
+```go
+func main() {
+	http.Handle("/", NewExampleRouter())
 
+	log.Println("Serving on port 8000")
+	err := http.ListenAndServe(":8000", nil)
+	if err != nil {
+		log.Fatalf("Server exited with: %v", err)
+	}
+}
 ```
-git clone https://github.com/windmilleng/servantes
-cd servantes; tilt up
+
+To start this server on Kubernetes, we need 3 configs:
+
+1) A [Dockerfile](https://github.com/windmilleng/tilt-example-go/blob/master/0-base/deployments/Dockerfile) that builds the image
+
+2) A [Kubernetes deployment](https://github.com/windmilleng/tilt-example-html/blob/master/0-base/deployments/kubernetes.yaml) that runs the image
+
+3) And finally, a Tiltfile that ties them together
+
+```python
+docker_build('example-go-image', '.', 
+    dockerfile='deployments/Dockerfile')
+k8s_yaml('deployments/kubernetes.yaml')
+k8s_resource('example-go', port_forwards=8000)
 ```
 
-Each widget is implemented by a different microservice backend.
+The first line tells Tilt to build an image with the name `example-html-image`
+in the directory `.` (the current directory).
+
+The second line tells Tilt to load the Kubernetes
+[Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#creating-a-deployment)
+yaml. The image name in the `docker_build` call must match the container `image`
+reference in the `example-go` Deployment.
+
+The last line configures port-forwarding so that your server is
+reachable at http://localhost:8000/. The resource name in the `k8s_resource` call
+must match the Deployment's `metadata.name` in `kubernetes.yaml`.
+
+Try it! Run:
+
+```
+git clone https://github.com/windmilleng/tilt-example-go
+cd tilt-example-go/0-base
+tilt up
+```
+
+Tilt will open a browser showing the web UI, a unified view that shows you app
+status and logs. Your terminal will also turn into a status box if you'd like to
+watch your server come up there.
+
+When it's ready, you will see the status icon turn green. The logs in the
+botton pane will display "Serving files on port 8000."
+
+<figure>
+  <a class="is-image" href="https://cloud.tilt.dev/snapshot/AZzpx-YL_-WxDmRFwlk=">
+    <img src="assets/docimg/example-go-image-1.png">
+  </a>
+  <figcaption>The server is up! Click the screenshot to see an interactive snapshot.</figcaption>
+</figure>
+
+## Step 1: Let's Add Benchmark Trickery
+
+Before we try to make this faster, let's measure it.
+
+Tilt can run commands locally, so that you can integrate your existing
```

Comment:
replace with [new copy from the template](https://github.com/windmilleng/tilt.specs/pull/27#pullrequestreview-373643423)